Userland/Applications/Browser/Tab.cpp

```diff
@@ -393,6 +395,29 @@ Tab::Tab(BrowserWindow& window)
     };
 }
 
+Optional<URL> Tab::read_url(bool can_append_extension)
+{
+    if (m_location_box->text().starts_with('?') && g_search_engine.is_empty()) {
+        GUI::MessageBox::show(&this->window(), "Select a search engine in the Settings menu before searching.", "No search engine selected", GUI::MessageBox::Type::Information);
+        return {};
+    }
+
+    const String& text = m_location_box->text();
```

Comment:
`text()` here returns a `String`, so it's mildly misleading to store it in a `String const&`, even if the language does allow it via const reference lifetime extension.